File: src/ch1/caesar_enc.py
def encrypt(text, shift):
    code_a = ord("a")
    code_z = ord("z")

    result = ""
    for ch in text:
        code_ch = ord(ch)
        if code_a <= code_ch <= code_z:
            code_ch = (code_ch - code_a + shift) % 26 + code_a
            if code_ch > code_z:
                code_ch = code_ch - 26
        result += chr(code_ch)

    return result


enc = encrypt("hello, world!", 3)
dec = encrypt(enc, -3)
print(enc, "=>", dec)

File: src/ch1/caesar_enc.rs
fn encrpt(text: &str, shift: i16) -> String {
    let code_a = 'a' as i16;
    let code_z = 'z' as i16;
    let mut result = String::new();
    for c in text.chars() {
        let mut code = c as i16;
        if code_a <= code && code <= code_z {
            code = (code - code_a + shift) % 26 + code_a;
        }
        result.push(code as u8 as char);
    }
    result
}

fn main() {
    let text = "hello, world!";
    let shift = 3;
    let enc = encrpt(text, shift);
    let dec = encrpt(&enc, -shift);
    println!("{} => {}", enc, dec);
}
